#!/usr/bin/env python
PACKAGE = 'navi_jackal'
import roslib;roslib.load_manifest(PACKAGE)
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
gen.add('F', double_t, 1, 'feedforward constant',  0.0)
gen.add('P', double_t, 1, 'proportional constant', 0.0)
gen.add('I', double_t, 1, 'integral constant',     0.0)
gen.add('D', double_t, 1, 'derivative constant',   0.0)
gen.add('threshold', int_t, 1, 'maximum allowable error', 0)

gen.add('ticks_per_rev', int_t,    2, 'encoder ticks per revolution', 800)
gen.add('pid_frequency', int_t,    2, 'control loop frequency in Hz', 50)
gen.add('pwm_max',       int_t,    2, 'maximum PWM value',            255)
gen.add('wheel_radius',  double_t, 2, 'wheel radius in meters',       0.127)
gen.add('robot_radius',  double_t, 2, 'robot radius in meters',       0.381)
gen.add('tau', double_t, 2, 'fit parameter of plant model', 0.0)
gen.add('k',   double_t, 2, 'fit parameter of plant model', 1.0)

exit(gen.generate(PACKAGE, 'navi_jackal', 'Jackal'))
